crates/task-impls/src/view_sync.rs

```diff
@@ -945,110 +1103,230 @@ impl<
             Leaf = Leaf<TYPES>,
             ConsensusMessage = SequencingMessage<TYPES, I>,
         >,
-    > ViewSyncRelayTaskState<TYPES, I>
-where
-    ViewSyncEx<TYPES, I>: ViewSyncExchangeType<
+    >
+    ViewSyncRelayTaskState<
```

Comment:
Just so I don't miss anything in the review: we basically have 3 copies of this task for each vote + cert pair?

Reply:
Yes, that is correct.  All this task does is collect votes into a certificate, so we can probably replace it with the generic vote collection task we talked about.